integrationTests/relayers/slowTests/framework/ethereumHandler.go

```diff
@@ -277,14 +277,22 @@ func (handler *EthereumHandler) IssueAndWhitelistToken(ctx context.Context, para
 	handler.checkEthTxResult(ctx, tx.Hash())
 
 	if len(params.InitialSupplyValue) > 0 {
-		if params.IsMintBurnOnEth {
-			mintAmount, ok := big.NewInt(0).SetString(params.InitialSupplyValue, 10)
-			require.True(handler, ok)
+		initialSupply, ok := big.NewInt(0).SetString(params.InitialSupplyValue, 10)
+		require.True(handler, ok)
 
-			tx, err = handler.SafeContract.InitSupplyMintBurn(auth, erc20Address, mintAmount, zeroValueBigInt)
-			require.NoError(handler, err)
+		if params.IsMintBurnOnEth {
+			var tx2 *types.Transaction
+			var err2 error
+			if params.IsNativeOnEth {
+				burnAmount := initialSupply
+				tx2, err2 = handler.SafeContract.InitSupplyMintBurn(auth, erc20Address, zeroValueBigInt, burnAmount)
+			} else {
+				mintAmount := initialSupply
+				tx2, err2 = handler.SafeContract.InitSupplyMintBurn(auth, erc20Address, mintAmount, zeroValueBigInt)
+			}
```

Comment:
perhaps extract `txInitSupplyMintBurn, errInitSupplyMintBurn := handler.SafeContract.InitSupplyMintBurn(auth, erc20Address, mintAmount, burnAmount)` to be called only once after this if and only set minAmount and burnAmount here